core/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java

```diff
@@ -72,6 +112,72 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String fieldInQuery = this.fieldName;
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            fieldInQuery = fieldType.names().indexName();
+        }
+        SpanQuery innerQuery = (SpanQuery) ((QueryBuilder) this.queryBuilder).toQuery(parseContext);
```

Comment:
Scratch that previous comment, I see what you mean now, will see where that comes from on master.